src/scaffoldmaker/meshtypes/meshtype_3d_lung3.py

```diff
@@ -0,0 +1,936 @@
+
+            if lung == leftLung:
+                octant_group_lists = []
+                for octant in range(8):
+                    octant_group_list = []
+                    octant_group_list.append(lungGroup.getGroup())
+                    octant_group_list.append(leftLungGroup.getGroup())
+                    octant_group_list.append((leftMedialLungGroup if (octant & 1) else leftLateralLungGroup).getGroup())
+                    octant_group_list.append((leftBaseLungGroup if (octant & 2) else leftPosteriorLungGroup).getGroup())
+                    if numberOfLeftLung > 1:
+                        octant_group_list.append((upperLeftLungGroup if (octant & 4) else lowerLeftLungGroup).getGroup())
+                    octant_group_lists.append(octant_group_list)
+            else:
+                octant_group_lists = []
+                for octant in range(8):
+                    octant_group_list = []
+                    octant_group_list.append(lungGroup.getGroup())
+                    octant_group_list.append(rightLungGroup.getGroup())
+                    octant_group_list.append((rightLateralLungGroup if (octant & 1) else rightMedialLungGroup).getGroup())
+                    octant_group_list.append((rightBaseLungGroup if (octant & 2) else rightPosteriorLungGroup).getGroup())
+                    if octant & 4:
+                        octant_group_list.append((middleRightLungGroup if (octant & 2) else upperRightLungGroup).getGroup())
+                    else:
+                        octant_group_list.append(lowerRightLungGroup.getGroup())
+                    octant_group_lists.append(octant_group_list)
+
+            ellipsoid.set_octant_group_lists(octant_group_lists)
+            annotationGroups += [boxGroup, transitionGroup]
+            ellipsoid.set_box_transition_groups(boxGroup.getGroup(), transitionGroup.getGroup())
+
+            ellipsoid.build()
+            nodeIdentifier, elementIdentifier = ellipsoid.generate_mesh(fieldmodule, coordinates, startNodeIdentifier, startElementIdentifier)
+
+            startNodeIdentifier, startElementIdentifier = nodeIdentifier, elementIdentifier
+
+        for lung in lungs:
+            isLeft = True if lung == leftLung else False
+            lungNodeset = leftLungNodesetGroup if isLeft else rightLungNodesetGroup
+            spacing = -lungSpacing if lung == leftLung else lungSpacing
+            zOffset = -0.5 * ellipsoid_height
+            lungMedialCurvature = -leftLungMedialCurvature if isLeft else leftLungMedialCurvature
+            rotateLungAngleY = rotateLeftLungY if isLeft else -rotateLeftLungY
+            rotateLungAngleZ = rotateLeftLungZ if isLeft else -rotateLeftLungZ
+
+            if edgeSharpFactor != 0.0:
+                taperLungEdge(edgeSharpFactor, fieldmodule, coordinates, lungNodeset, halfBreadth)
+
+            if baseSharpFactor != 0.0:
+                taperLungEdge(baseSharpFactor, fieldmodule, coordinates, lungNodeset, halfHeight, isBase=True)
```

Comment:
I'm suggesting different taper functions (see later) that give zero displacement and rate of change of displacement at the start y (or z) increasing to maximum at end y (or z) which is placed just past the end of the ellipsoid. These are some suggested values, but the values should be tuned:
```
            if edgeSharpFactor != 0.0:
                taperLungEdgeY(edgeSharpFactor, fieldmodule, coordinates, lungNodeset, -0.5 * halfBreadth,
                               1.1 * halfBreadth)

            if baseSharpFactor != 0.0:
                taperLungEdgeZ(baseSharpFactor, fieldmodule, coordinates, lungNodeset, 0.5 * halfHeight,
                               -1.1 * halfHeight)
```